tests/ProductTest.php

```diff
@@ -1464,6 +1464,47 @@ public function testProcessVariationSpecificProperties($previouslyParsedData, $d
         $this->assertSame($expectedResult, $productMock->getField('attributes'));
     }
 
+    /**
+     * @test
+     */
+    public function testProcessVariationSpecificPropertiesReturnsWithoutProcessingIfPropertiesRegistryReturnsFalse()
+    {
+        $registryMock = $this->getMockBuilder(\Findologic\Plentymarkets\Registry::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['get'])
+            ->getMock();
+        $registryMock->expects($this->once())->method('get')->will($this->returnValue(false));
+
+        $productMock = $this->getProductMock(['handleEmptyData'], ['registry' => $registryMock]);
+        $productMock->expects($this->once())->method('handleEmptyData')->with('Variation properties are missing');
+
+        $productMock->processVariationSpecificProperties(['test' => 'test']);
```

Comment:
Add an assertion which calls `\Findologic\Plentymarkets\Product::getAttributeField` to make sure that the properties are indeed not set.